app/services/delete_account_service.rb

```diff
@@ -151,9 +151,7 @@ def purge_content!
   end
 
   def purge_statuses!
-    @account.statuses.reorder(nil).find_in_batches do |statuses|
-      statuses.reject! { |status| reported_status_ids.include?(status.id) } if keep_account_record?
-
+    @account.statuses.reorder(nil).where.not(id: reported_status_ids).in_batches do |statuses|
```

Comment:
The change is probably wrong here because we don't use `statuses` as a relation. If you want to iterate on the relation you have to call `each_record` according to the documentation which we do not do.

Reply:
`BatchEnumerator` is `Enumerable` so I don't think that should be an issue.

Reply:
It's not as obvious to me... The variable is accessed multiple times, what is the behaviour of this enumerable? What if each time something implicitly casts to_a on it, it requeries the database? Why do the [docs](https://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-in_batches) specify:

> NOTE: If you are going to iterate through each record, you should call each_record on the yielded BatchEnumerator:

Reply:
Hmm. I don't know. Well, I can change that, it doesn't appear to make a big difference (but a positive one nevertheless).